core/src/main/scala/kafka/log/remote/RemoteIndexCache.scala

```diff
@@ -266,13 +309,21 @@ class RemoteIndexCache(maxSize: Int = 1024, remoteStorageManager: RemoteStorageM
     getIndexEntry(remoteLogSegmentMetadata).lookupTimestamp(timestamp, startingOffset).position
   }
 
+  /**
+   * Close should synchronously cleanup the resources used by this cache.
+   * This index is closed when [[RemoteLogManager]] is closed.
+   */
   def close(): Unit = {
-    closed = true
-    cleanerThread.shutdown()
-    // Close all the opened indexes.
-    lock synchronized {
-      entries.values().stream().forEach(entry => entry.close())
+    // make close idempotent
+    if (!closed.getAndSet(true)) {
+      // Initiate shutdown for cleaning thread
+      val shutdownRequired = cleanerThread.initiateShutdown()
+      // Close all the opened indexes to force unload mmap memory. This does not delete the index files from disk.
+      internalCache.asMap().forEach((_, entry) => entry.close())
+      // Perform any pending activities required by the cache for cleanup
+      internalCache.cleanUp()
```

Comment:
this would do pending maintenance work like expiration. Perhaps you meant `invalidateAll()` to discard all of the entries?

Reply:
Great call out. I did not realise that. I do not want removal listener to execute on shutdown. I will not do anything at all here and just let GC clean it up.

Am I correct in assuming that no specific "cleanup" is necessary to release the resources acquired by the cache?

Reply:
Yep, it is just a fancy in-memory hash map. Caffeine doesn’t create threads, files, whatever. The cleanUp() method simply runs the maintenance work like discarding expired or weak/soft collected entries, which might occur before it realizes otherwise. It if fully gc’able.

Reply:
Thank you for jumping in this review and providing your expert opinion on this cache. Appreciate it!